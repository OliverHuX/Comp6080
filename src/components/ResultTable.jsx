
import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Title from './Title';
import PropTypes from 'prop-types';

export default function ResultTable ({ title = 'Table', data }) {
  return (
    <React.Fragment>
      <Title>{title}</Title>
      <Table size="small">
        <TableHead>
          <TableRow>
            <TableCell>Rank</TableCell>
            <TableCell>Name</TableCell>
            <TableCell>Answered Correct</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data && data.map((row) => (
            <TableRow key={row.rank}>
              <TableCell>{row.rank}</TableCell>
              <TableCell>{row.name}</TableCell>
              <TableCell>{row.score}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </React.Fragment>
  );
}
ResultTable.propTypes = {
  title: PropTypes.string,
  data: PropTypes.array,
};
