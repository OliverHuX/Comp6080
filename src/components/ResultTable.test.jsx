import React from 'react';
import ResultTable from './ResultTable';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Title from './Title';

describe('ResultTable Component', () => {
  const data = [
    { rank: 1, name: 'Alice', score: 100 },
    { rank: 2, name: 'Ben', score: 90 },
    { rank: 3, name: 'Cathy', score: 80 },
    { rank: 4, name: 'Dan', score: 70 },
    { rank: 5, name: 'Emily', score: 60 },
  ];
  const title = 'Rank Info';
  it('uses fallback title if none provided', () => {
    const resultTable = shallow(<ResultTable data={data} />);
    resultTable.contains(<Title>Table</Title>);
  });
  it('has the given title in the table', () => {
    const resultTable = shallow(<ResultTable data={data} title={title}/>);
    const tableTitle = shallow(resultTable.props().children[0]);
    expect(tableTitle.text()).toBe(title);
    resultTable.contains(<Title>{title}</Title>);
  });
  it('has the given data in the table', () => {
    const resultTable = shallow(<ResultTable data={data} title={title}/>);
    const tableContent = shallow(resultTable.props().children[1]);
    expect(tableContent.text()).toBe('RankNameAnswered Correct1Alice1002Ben903Cathy804Dan705Emily60');
  });
  it('has only table header when no data provided', () => {
    const resultTable = shallow(<ResultTable/>);
    const tableContent = shallow(resultTable.props().children[1]);
    expect(tableContent.text()).toBe('RankNameAnswered Correct');
  });
  it('renders with minimal props (snapshot)', () => {
    const resultTable = renderer.create(<ResultTable/>).toJSON();
    expect(resultTable).toMatchSnapshot();
  });
  it('renders with provided title with no data (snapshot)', () => {
    const resultTable = renderer.create(<ResultTable title={title}/>).toJSON();
    expect(resultTable).toMatchSnapshot();
  });
  it('renders with provided title and data (snapshot)', () => {
    const resultTable = renderer.create(<ResultTable data={data} title={title}/>).toJSON();
    expect(resultTable).toMatchSnapshot();
  });
  it('renders with default title and data (snapshot)', () => {
    const resultTable = renderer.create(<ResultTable data={data} />).toJSON();
    expect(resultTable).toMatchSnapshot();
  });
});
