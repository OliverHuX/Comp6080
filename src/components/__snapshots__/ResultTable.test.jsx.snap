// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultTable Component renders with default title and data (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Table
  </h2>,
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Rank
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Answered Correct
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Alice
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          100
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Ben
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          90
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Cathy
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          80
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          4
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Dan
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          70
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          5
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Emily
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          60
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`ResultTable Component renders with minimal props (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Table
  </h2>,
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Rank
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Answered Correct
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>,
]
`;

exports[`ResultTable Component renders with provided title and data (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Rank Info
  </h2>,
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Rank
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Answered Correct
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Alice
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          100
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Ben
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          90
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Cathy
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          80
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          4
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Dan
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          70
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          5
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Emily
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          60
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`ResultTable Component renders with provided title with no data (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Rank Info
  </h2>,
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Rank
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Answered Correct
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>,
]
`;
